.github/workflows/publish_layer.yml

```diff
@@ -0,0 +1,207 @@
+name: Deploy layer to all regions
+
+permissions:
+  id-token: write
+  contents: read
+
+on:
+  # TODO: remove after setup is finished
+  workflow_dispatch: { }
```

Comment:
Unless you truly need a explicit checkout from the `pull_request.head.ref`, you can also enable manual workflow runs - example: https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/workflows/rebuild_latest_docs.yml#L12

Omitting those will checkout HEAD and run its course normally. If you'd want to checkout from a specific point in time OR from a separate repository, then these would be needed.

Reply:
Adding input to `workflow_dispatch` is necessary because we need the version. Any ideas on how to assign one of the inputs to the `RELEASE_TAG_VERSION`? Need some bash wizardry without if-statements

```
RELEASE_TAG_VERSION=${{ github.event.release.tag_name }}  || ${{ inputs.latest_published_version }}
```

Not sure nested `$` works in yaml to assign default value 🤔

```
RELEASE_TAG_VERSION=${ ${{ github.event.release.tag_name }}:-${{ inputs.latest_published_version }} }
```